File: test/python/TestSpaceGroupCSV.py
#!/usr/bin/env python3

##  ***********************************************************************************************
##
##  OpenHKL: data reduction for single crystal diffraction
##
##! @file      test/python/TestSpaceGroupCSV.py
##! @brief     Test ...
##!
##! @homepage  https://openhkl.org
##! @license   GNU General Public License v3 or higher (see COPYING)
##! @copyright Institut Laue-Langevin and Forschungszentrum Jülich GmbH 2016-
##! @authors   see CITATION, MAINTAINER
##
##  ***********************************************************************************************

import pyohkl as ohkl
# import numpy as np
import unittest

class TestSpaceGroupCSV(unittest.TestCase):

    def test(self):
        num_rows = 0

        counts = map
        symbols = ohkl.SpaceGroup.symbols()
        for symbol in symbols:
            # syntax error
            #counts([symbol]) = 0
            pass


if __name__ == '__main__':
    unittest.main()

File: test/cpp/unit/crystal/TestSpaceGroupCSV.cpp
//  ***********************************************************************************************
//
//  OpenHKL: data reduction for single crystal diffraction
//
//! @file      test/cpp/crystal/TestSpaceGroupCSV.cpp
//! @brief     Test ...
//!
//! @homepage  https://openhkl.org
//! @license   GNU General Public License v3 or higher (see COPYING)
//! @copyright Institut Laue-Langevin and Forschungszentrum Jülich GmbH 2016-
//! @authors   see CITATION, MAINTAINER
//
//  ***********************************************************************************************

#include "test/cpp/catch.hpp"

#include "base/utils/CSV.h"
#include "tables/crystal/SpaceGroup.h"

#include <fstream>
#include <string>
#include <iostream>

TEST_CASE("test/crystal/TestSpaceGroupCSV.cpp", "")
{
    std::ifstream csv_file;
    csv_file.open("crystallography.tsv", std::ifstream::in);
    if (!csv_file.is_open()) {
        std::cerr << "failed to open crystallography.tsv\n";
        CHECK(false);
        return;
    }

    ohkl::CSV csv_reader('\t', '#');

    while (!csv_file.eof()) {
        std::vector<std::string> row = csv_reader.getRow(csv_file);

        // extra newline at end of file
        if (csv_file.eof() && row.size() < 8)
            continue;

        CHECK(row.size() == 8);

        std::cout << row[0] << std::endl;

        CHECK_NOTHROW(ohkl::SpaceGroup{row[0]});
    }
}
